src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+            const auto is_right_turn = road_index < *obvious;
+            const auto is_left_turn = road_index > *obvious;
+
+            const auto &extractor = intersection_generator.GetCoordinateExtractor();
+
+            const NodeID start = intersection_node_id; // b
+            const EdgeID edge = sliproad_edge;         // bd
+
+            const auto coords = extractor.GetForwardCoordinatesAlongRoad(start, edge);
+            BOOST_ASSERT(coords.size() >= 2);
+
+            // Now keep start and end coordinate fix and check for curvature
+            const auto start_coord = coords.front();
+            const auto end_coord = coords.back();
+
+            const auto first = std::begin(coords) + 1;
+            const auto last = std::end(coords) - 1;
+
+            auto snuggles = false;
+
+            using namespace util::coordinate_calculation;
+
+            // In addition, if it's a right/left turn we expect the rightmost/leftmost
+            // turn at `c` to be more or less ~90 degree for a Sliproad scenario.
+            double deviation_from_straight = 0;
+
+            if (is_right_turn)
+            {
+                snuggles = std::all_of(first, last, [=](auto each) { //
+                    return !isCCW(start_coord, each, end_coord);
+                });
+
+                const auto rightmost = next->intersection.getRightmostRoad();
+                deviation_from_straight = angularDeviation(rightmost.angle, STRAIGHT_ANGLE);
+            }
+            else if (is_left_turn)
+            {
+                snuggles = std::all_of(first, last, [=](auto each) { //
+                    return isCCW(start_coord, each, end_coord);
+                });
+
+                const auto leftmost = next->intersection.getLeftmostRoad();
+                deviation_from_straight = angularDeviation(leftmost.angle, STRAIGHT_ANGLE);
+            }
+
+            // The data modelling for small Sliproads is not reliable enough.
+            // Only check for curvature and ~90 degree when it makes sense to do so.
+            const constexpr auto MIN_LENGTH = 3.;
+
+            const auto length = haversineDistance(node_info_list[intersection_node_id], //
+                                                  node_info_list[next->node]);
+
+            const double perpendicular_angle = 90 + FUZZY_ANGLE_DIFFERENCE;
+
+            if (length >= MIN_LENGTH)
+            {
+                if (!snuggles)
+                    continue;
+
+                if (deviation_from_straight > perpendicular_angle)
+                    continue;
+            }
+        }
+
+        // Check for area under triangle `bdc`.
+        //
+        // a ... b .... c .
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d
+        //
+        const auto area_threshold = std::pow(
+            scaledThresholdByRoadClass(MAX_SLIPROAD_THRESHOLD, sliproad_data.road_classification),
```

Comment:
assuming the length of the sliproad scales linearly, I'd expect `scaledThreshold()/2` here to have a more consistent relation between sliproadLength (which is the threshold, right?) and triangle are.